### PR TITLE
Fix few bugs

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

File: packages/dev/core/src/Maths/math.vector.ts [hunk 1]
```diff
@@ -6480,6 +6480,8 @@ export class Matrix {
         m[14] = 0;
         m[15] = 1;
         result.markAsUpdated();
+
+        console.log(result.determinant());
```
Comment (sebavan) [inline packages/dev/core/src/Maths/math.vector.ts:6484]: :-)

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Maths/math.vector.ts [hunk 2]
```diff
@@ -6445,18 +6445,18 @@ export class Matrix {
         const m = result._m;
         if (c < -1 + Epsilon) {
             // from and to are colinear and opposite direction.
-            // compute a PI rotation on Z axis
+            // compute a PI rotation on Y axis
```
Comment (sebavan) [inline packages/dev/core/src/Maths/math.vector.ts:6448]: Is this not a breaking change ???

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14282/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14282/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14282/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14282/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14282/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14282/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14282/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14282/merge#BCU1XR#0

**Pull request merged**